android/build.gradle

```diff
@@ -46,6 +46,11 @@ android {
     compileSdkVersion getExtOrDefault('compileSdkVersion').toInteger()
     buildToolsVersion getExtOrDefault('buildToolsVersion')
 
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
```

Comment:
I don't think we should upgrade to Java 8 at the moment.
Let's revert back to the original version we were using (I think this means we will also need to replace lambdas with anonymous classes and probably other changes).
I definitely think we should take this decision of just using Java 8 and migrate to new features like lambda expressions some day but we shouldn't do that on the fly as part of another PR without carefully evaluating it.